src/Domain/LeanCode.TimeProvider/AsyncLocalTimeProvider.cs

```diff
@@ -0,0 +1,40 @@
+namespace LeanCode.TimeProvider;
+
+public sealed class AsyncLocalTimeProvider(System.TimeProvider fallback) : System.TimeProvider
+{
+    public static AsyncLocalTimeProvider SystemSharedInstance { get; } = new(System);
```

Comment:
I suppose this one won't be of much use to us so I could get rid of it and keep only `UtcSystemSharedInstance`.

Reply:
IMO, calling it `TestTimeProvider` (fixed no longer fits), having only one static instance, and making `Activate` method static would make better dx for our use case.

Reply:
Right, that would also make it closer to previous `FixedTimeProvider.SetTo`, and having multiple shared static instances doesn't really work anyway; you could pick one to use in your project's tests but then you can never use the other because `Time` only has a single reference so they would evict each another.

On the other hand, being more explicit about which instance we want to use is imo a good thing, even if there's most commonly only one: `TestTimeProvider.Shared.Activate` isn't much longer than `TestTimeProvider.Activate` and doesn't favor our single instance that much… but maybe it should? :thinking:

I'm indifferent about the name, one explains the behavior and the other explains intent. Both work for me but let's see what @jakubfijalkowski says about this.

Reply:
> but maybe it should?

IMO it should. "Shared" instance, where you really don't need shared but a singleton, hints that there might be a case for more instances. Here, _really_ static instance makes for a better interface IMO.